### DISCUSSION TITLE
client.on event 질문

### DISCUSSION THREAD

**Discussion by disabled-variable** [storycraft/node-kakao#682] [Q&A] [answered]
이벤트를 주지않고 로그인하면 자동적으로 실행되도록 하는 방법이 있을까요?
 
예) 카카오톡에 로그인 성공하면, 즉시 본계정에 '로그인되었습니다'라고 메시지를 전송..

**Comment by MoonJR** [comment, marked as answer]
const memoChat = Array.from(talkClient.channelList.normal.all());

이런식으로 채팅방 목록 가져와서 메모챗 필터링 해서 보내시면 될것 같긴하네요.

**Reply by disabled-variable** [reply]
<ref *2> TalkNormalChannel {
    _events: Events <[Object: null prototype] {}> {},
    _eventsCount: 0,
    _channel: { channelId: new Long("319264624739608"), info: [Object] },
    _chatListStore: TalkMemoryChatListStore { limit: 300, _chatList: [] },
    _channelSession: TalkChannelDataSession {
      _clientUser: [Object],
      _channelSession: [TalkChannelSession],
      _store: [TalkMemoryChannelDataStore],
      _chatListStore: [TalkMemoryChatListStore]
    },
    _normalChannelSession: TalkNormalChannelDataSession {
      _clientUser: [Object],
      _channelSession: [TalkNormalChannelSession],
      _store: [TalkMemoryChannelDataStore]
    },
    _handler: TalkChannelHandler {
      _channel: [Circular *2],
      _emitter: [Circular *2],
      _store: [TalkMemoryChannelDataStore],
      _chatListStore: [TalkMemoryChatListStore]
    },
    _normalHandler: TalkNormalChannelHandler {
      _channel: [Circular *2],
      _session: [TalkNormalChannelSession],
      _emitter: [Circular *2],
      _store: [TalkMemoryChannelDataStore],
      _chatListStore: [TalkMemoryChatListStore]
    }
  }

출력값이 이렇게 나오는데 channelID 기준으로 채팅을 전송하면 되겠죠?
혹시 channelID 값만을 출력할 순 없을까요?

**Reply by generalxd** [reply]
로그인 하셨으면 그뒤에 출력하는 부분을 넣으면됩니다

CLIENT.clientUser 인가 거기에 기본 로그인한 사람의 정보는 다들어있어요
